test/extended/prometheus/collection_profiles.go

```diff
@@ -0,0 +1,312 @@
+		}
+		r.mclient, err = prometheusoperatorv1client.NewForConfig(oc.AdminConfig())
+		if err != nil {
+			g.Fail(fmt.Sprintf("failed to create monitoring client: %v", err))
+		}
+		r.pclient = oc.NewPrometheusClient(tctx)
+
+		var operatorConfiguration *v1.ConfigMap
+		o.Eventually(func() error {
+			operatorConfiguration, err = r.kclient.CoreV1().ConfigMaps(operatorNamespaceName).Get(tctx, operatorConfigurationName, metav1.GetOptions{})
+			return err
+		}).Should(o.BeNil())
+		r.originalOperatorConfiguration = operatorConfiguration
+	})
+
+	g.AfterAll(func() {
+		currentConfiguration, err := r.kclient.CoreV1().ConfigMaps(operatorNamespaceName).Get(tctx, operatorConfigurationName, metav1.GetOptions{})
+		o.Expect(err).To(o.BeNil())
+		currentConfiguration.Data = r.originalOperatorConfiguration.Data
+		_, err = r.kclient.CoreV1().ConfigMaps(operatorNamespaceName).Update(tctx, currentConfiguration, metav1.UpdateOptions{})
+		o.Expect(err).To(o.BeNil())
+	})
+
+	var _ = g.Context("in an environment that migrates from default to minimal collection profile", func() {
+		g.It("should apply default collection profile initially", func() {
+			o.Eventually(func() error {
+				monitors, err := r.fetchMonitorsFor([2]string{collectionProfileFeatureLabel, collectionProfileDefault})
+				if err != nil {
+					return err
+				}
+				if len(monitors.Items) == 0 {
+					return fmt.Errorf("no monitors found with collection profile %q", collectionProfileDefault)
+				}
+
+				return nil
+			}).Should(o.BeNil())
+		})
+		g.It("should expose information about the applied collection profile using meta-metrics", func() {
+			vectorExpression := "profile:cluster_monitoring_operator_collection_profile:max{profile=\"%s\"} == 1"
+
+			for _, profile := range collectionProfilesSupportedList {
+				err := r.makeCollectionProfileConfigurationFor(tctx, profile)
+				o.Expect(err).To(o.BeNil())
+
+				o.Eventually(func() error {
+					queryResponse, err := helper.RunQuery(tctx, r.pclient, fmt.Sprintf(vectorExpression, profile))
+					if err != nil {
+						return err
+					}
+					if len(queryResponse.Data.Result) == 0 {
+						return fmt.Errorf("no result found for profile %q", profile)
+					}
+
+					return nil
+				}).Should(o.BeNil())
+			}
+		})
+		g.It("should have at least one implementation for each collection profile", func() {
+			for _, profile := range collectionProfilesSupportedList {
+				err := r.makeCollectionProfileConfigurationFor(tctx, profile)
+				o.Expect(err).To(o.BeNil())
+
+				o.Eventually(func() error {
+					monitors, err := r.fetchMonitorsFor([2]string{collectionProfileFeatureLabel, profile})
+					if err != nil {
+						return err
+					}
+					if len(monitors.Items) == 0 {
+						return fmt.Errorf("no monitors found with collection profile %q", profile)
+					}
+
+					return nil
+				}).Should(o.BeNil())
+			}
+		})
+	})
+
+	var _ = g.Context("in an environment that has the minimal profile pre-set", func() {
```

Comment:
I have to again point out I'm not a ginko expert, but shouldn't this context have a `BeforeAll` node to apply the minimal profile? Iiuc the intention is that ginko can reorder context nodes as it sees fit, so I'm not sure this will succeed in all cases. This currently relies on the previous context to apply the minimal profile.

Reply:
The [documentation](https://pkg.go.dev/github.com/onsi/ginkgo/v2@v2.19.1#Describe) mentions that `Context` ([alias for `Describe`](https://github.com/onsi/ginkgo/blob/224be5bf5590e4159b19e750f013cb77d68f2609/dsl/core/core_dsl.go#L47)) supports decorators (i.e., the `Ordered` parameter being passed here) while nesting more containers internally. The containers execute [as they appear](https://github.com/onsi/ginkgo/blob/v2.19.1/internal/ordering.go#L52) (among other "weights" that work towards the same affect, even [`ID`s](https://github.com/onsi/ginkgo/blob/v2.19.1/internal/node.go#L22), in case of `It`s).

As long as the structure is maintained, things should happen as expected. But LMK if this seems prone to manual oversight and I'll add those in.

Reply:
The [documentation](https://pkg.go.dev/github.com/onsi/ginkgo/v2@v2.19.1#Describe) mentions that `Context` ([alias for `Describe`](https://github.com/onsi/ginkgo/blob/224be5bf5590e4159b19e750f013cb77d68f2609/dsl/core/core_dsl.go#L47)) supports decorators (i.e., the `Ordered` parameter being passed here) while nesting more containers internally. The containers seem to execute [as they appear](https://github.com/onsi/ginkgo/blob/v2.19.1/internal/ordering.go#L52) (among other "weights" that work towards the same affect, and [`ID`s](https://github.com/onsi/ginkgo/blob/v2.19.1/internal/node.go#L22), in case of `It`s).

I believe that as long as the structure is maintained, things should happen as expected. But LMK if this seems prone to manual oversight and I'll add those in.

Reply:
Ahh I missed that. Should we maybe have a comment preceding the context so future readers don't miss this like I did?